src/staking-v3/logic/services/DappStakingServiceEvm.ts

```diff
@@ -0,0 +1,244 @@
+import { inject, injectable } from 'inversify';
+import { IDappStakingService } from './IDappStakingService';
+import { DappStakingService } from './DappStakingService';
+import { DappStakeInfo } from '../models';
+import { IWalletService } from '../../../v2/services/IWalletService';
+import { IDappStakingRepository } from '../repositories';
+import { Symbols } from 'src/v2/symbols';
+import { evmPrecompiledContract } from 'src/modules/precompiled';
+import { IAccountUnificationRepository } from 'src/v2/repositories';
+import { Guard } from 'src/v2/common';
+
+const { dispatch } = evmPrecompiledContract;
+
+@injectable()
+export class DappStakingServiceEvm extends DappStakingService implements IDappStakingService {
+  private readonly wallet: IWalletService;
+
+  constructor(
+    @inject(Symbols.DappStakingRepositoryV3) dappStakingRepository: IDappStakingRepository,
+    @inject(Symbols.WalletFactory) walletFactory: () => IWalletService,
+    @inject(Symbols.AccountUnificationRepository),
+    private accountUnificationRepository: IAccountUnificationRepository
+  ) {
+    super(dappStakingRepository, walletFactory);
+    this.wallet = walletFactory();
+  }
+
+  // @inheritdoc
+  public async claimUnstakeAndUnlock(
+    contractAddress: string,
+    amount: number,
+    senderAddress: string,
+    successMessage: string
+  ): Promise<void> {
+    const ss58Address = await this.getSS58Address(senderAddress);
+    Guard.ThrowIfUndefined(contractAddress, 'contractAddress');
+    Guard.ThrowIfUndefined(senderAddress, 'senderAddress');
+    Guard.ThrowIfNegative('amount', amount);
+
+    const claimStakerCalls = await this.getClaimStakerAndBonusRewardsCalls(ss58Address);
```

Comment:
This code block is the same as in native service. I suggest to refactor this to avoid code duplication. See `claimLockAndStake` method for example. Basically extract duplicated code block to protected method and put it to native service.
Do similar for other calls if there is repeating code.

Reply:
Sorry but I'm afraid I don't understand your point. 'claimStakerCalls' is required for the batch call, therefore I don't think we can make it more simpler here.

Reply:
Are you suggesting me to move the logic into `DappStakingRepository` and then refactor both 'DappsStakingServiceEvm' and 'DappStakingService'?

Reply:
Ok, will do that

Reply:
Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/1102/commits/9ef93188ebd7d804e4b115a211cea97a7cfb29c5) commit.